test_pipeline.py

```diff
@@ -0,0 +1,149 @@
+        production_csv_drive_path = args.drive_upload[3]
+
+    user = args.user
+
+    phone_number_uuid_table_path = args.phone_number_uuid_table_path
+    water_filter_input_path = args.water_filter_input_path
+    internet_working_input_path = args.internet_working_input_path
+    waste_disposal_input_path = args.waste_disposal_input_path
+    demog_input_path = args.demog_input_path
+    prev_coded_dir_path = args.prev_coded_dir_path
+
+    json_output_path = args.json_output_path
+    icr_output_dir = args.icr_output_dir
+    coded_dir_path = args.coded_dir_path
+    csv_by_message_output_path = args.csv_by_message_output_path
+    csv_by_individual_output_path = args.csv_by_individual_output_path
+    production_csv_output_path = args.production_csv_output_path
+
+    shows_paths = [water_filter_input_path, internet_working_input_path, waste_disposal_input_path]
+
+    # Load phone number <-> UID Table
+    print("Loading phone Number <-> UUID Table...")
+    with open(phone_number_uuid_table_path, "r") as f:
+        phone_number_uuid_table_path = PhoneNumberUuidTable.load(f)
+
+    # Load Messages
+    shows_datasets = []
+    for i, path in enumerate(shows_paths):
+        print("loading Show {}/{}...".format(i + 1, len(shows_paths)))
+        with open(path, "r") as f:
+            shows_datasets.append(TracedDataJsonIO.import_json_to_traced_data_iterable(f))
+
+    # Load surveys
+    print("Loading Demographics...")
+    with open(demog_input_path, "r") as f:
+        demographics = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # Add survey data to the shows
+    print("Combining shows-datasets & demogs...")
+    data = CombineRawDatasets.combine_raw_datasets(user, shows_datasets, [demographics])
+
+    print("Translating Rapid Pro Keys...")
+    data = TranslateRapidProKeys.translate_rapid_pro_keys(user, data, prev_coded_dir_path)
+
+    print("Exporting production CSV...")
+    data = ProductionFile.generate(data, production_csv_output_path)
+    
+    print("Writing TracedData to file....")
+    IOUtils.ensure_dirs_exist_for_file(json_output_path)
+    with open(json_output_path, "w") as f:
+        TracedDataJsonIO.export_traced_data_iterable_to_json(data, f, pretty_print=True)
+    
+    if drive_upload:
+        print("Uploading CSVs to Google Drive...")
+        drive_client_wrapper.init_client(drive_credentials_path)
+
+        production_csv_drive_dir = os.path.dirname(production_csv_drive_path)
+        production_csv_drive_file_name = os.path.basename(production_csv_drive_path)
+        drive_client_wrapper.update_or_create(production_csv_output_path, production_csv_drive_dir,
+                                            target_file_name=production_csv_drive_file_name,
+                                            target_folder_is_shared=True)
+        print("Files successfully uploaded")
+
+    else:
+        print("Not uploading to Google Drive")
+    
+    print("Writing TracedData to file....")
```

Comment:
I know I did it in this order on ADSS, but I am starting to think we should write the TracedData before the Drive upload. The problem with the order used here is that if writing the TracedData fails, then we have analysis files in Drive but no TracedData associated with them...

Reply:
That makes sense thank @as2388